internal/controllers/gateway/route_utils.go

```diff
@@ -195,32 +196,43 @@ func getSupportedGatewayForRoute[T gatewayapi.RouteT](ctx context.Context, mgrc
 			allowedByAllowedRoutes  = false
 			allowedBySupportedKinds = false
 			allowedByListenerName   = false
+			listenerReady           = false
 		)
 
 		for _, listener := range gateway.Spec.Listeners {
+			listenerLogger := gwLogger.WithValues("listener", string(listener.Name))
 			// Check if the route matches listener's AllowedRoutes.
 			if ok, err := routeMatchesListenerAllowedRoutes(ctx, mgrc, route, listener, gateway.Namespace, parentRef.Namespace); err != nil {
 				return nil, fmt.Errorf("failed matching listener %s to a route %s for gateway %s: %w",
 					listener.Name, route.GetName(), gateway.Name, err,
 				)
 			} else if !ok {
+				listenerLogger.V(util.DebugLevel).Info("Route does not match listener's allowed routes")
 				continue
 			}
 			allowedByAllowedRoutes = true
 
 			// Check the listeners statuses:
 			// - Check if a listener status exists with a matching type (via SupportedKinds).
 			// - Check if it matches the requested listener by name (if specified).
-			// - And finally check if that listeners is marked as Ready.
-			if err := existsMatchingReadyListenerInStatus(route, listener, gateway.Status.Listeners); err != nil {
+			if err := existsMatchingListenerInStatus(route, listener, gateway.Status.Listeners); err != nil {
+				listenerLogger.V(util.DebugLevel).Info("Listener does not support this route", "reason", err.Error())
 				continue
 			} else { //nolint:revive
 				allowedBySupportedKinds = true
 			}
 
+			if err := listenerProgrammedInStatus(listener.Name, gateway.Status.Listeners); err != nil {
+				listenerLogger.V(util.DebugLevel).Info("Listener is not ready", "reason", err.Error())
+				continue
+			} else { //nolint:revive
+				listenerReady = true
+			}
```

Comment:
```suggestion
			}
			listenerReady = true
```
I think it is equivalent and we don't have `//nolint:` the same can be done for lines 221:223

Reply:
While I'd normally agree with this I suggest that we keep it this way to make the hint to the reader that this is not always as in this `if`: in other "checks" below and above we might check for e.g. `!ok`. We don't want to set it regardless outside of the `if` (e.g. in lines 209-211).

Let's be consistent. We might refactor this ares of the codebase if need be separately.

Reply:
It can be left as it is. For me it's rather not a hint it's cognitive load, because something unexpected happens. For me to be consistent I'd suggest to have
```
			// Check if the route matches listener's AllowedRoutes.
			ok, err := routeMatchesListenerAllowedRoutes(ctx, mgrc, route, listener, gateway.Namespace, parentRef.Namespace)
			if err != nil {
				return nil, fmt.Errorf("failed matching listener %s to a route %s for gateway %s: %w",
					listener.Name, route.GetName(), gateway.Name, err,
				)
			}
			if !ok {
				listenerLogger.V(util.DebugLevel).Info("Route does not match listener's allowed routes")
				continue
			}
```
and in 266 revert the conditional. 

But it's ok to leave it as it is and :shipit: